bugbug/models/bugtype.py

```diff
@@ -0,0 +1,133 @@
+
+
+class BugTypeModel(BugModel):
+    def __init__(self, lemmatization=False, historical=False):
+        BugModel.__init__(self, lemmatization)
+
+        self.sampler = BorderlineSMOTE(random_state=0)
+
+        keywords_to_ignore = {
+            "sec-critical",
+            "sec-high",
+            "sec-moderate",
+            "sec-low",
+            "sec-other",
+            "sec-audit",
+            "sec-vector",
+            "sec-want",
+            "memory-footprint",
+            "memory-leak",
+            "crash",
+            "crashreportid",
+            "perf",
+        }
+
+        feature_extractors = [
+            bug_features.has_str(),
+            bug_features.severity(),
+            # Ignore keywords that would make the ML completely skewed
+            # (we are going to use them as 100% rules in the evaluation phase).
+            bug_features.keywords(keywords_to_ignore),
+            bug_features.is_coverity_issue(),
+            bug_features.has_crash_signature(),
+            bug_features.has_url(),
+            bug_features.has_w3c_url(),
+            bug_features.has_github_url(),
+            bug_features.whiteboard(),
+            bug_features.patches(),
+            bug_features.landings(),
+            bug_features.title(),
+            bug_features.blocked_bugs_number(),
+            bug_features.ever_affected(),
+            bug_features.affected_then_unaffected(),
+            bug_features.product(),
+            bug_features.component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.url(),
+            feature_cleanup.fileref(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = OneVsRestClassifier(xgboost.XGBClassifier(n_jobs=16))
+
+    def get_labels(self):
+        classes = {}
+        keyword_list = ["performance", "memory", "crash", "security"]
+
+        def bug_type(keyword):
+            keyword_dict = {
```

Comment:
Define this keyword_dict as a global variable at the beginning of the file, so you can also use it instead of keywords_to_ignore (and avoid defining the same keywords multiple times).